week-1-ts-ecosystem/w01e04-light-me-up/src/modules/powerSourceModule.ts

```diff
@@ -0,0 +1,12 @@
+export class PowerSource {
+  private energySupply: number = 100;
+  
+  consume(energy: number) {
+    if (this.energySupply <= 0)
+      throw new Error("Power source is depleted to zero");
+    else {
+        this.energySupply -= energy;
```

Comment:
Tutaj rozumiem Twój tok myślenia - że sobie zabierzemy energię a potem,
jak ktoś następny będzie jej potrzebował - to dopiero rzucimy wyjątek że "już nie ma"

Jednak tutaj jest pewien błąd logiczny - ponieważ Twój kod pozwala pobrać ze źródła więcej mocy niż jest dostępne.
sprawiając że `energySupply` może być wartością na `-` 
taka sytuacja nie powinna mieć miejsca.

Rzucamy wyjątek (ten sam) - już w momencie w którym pobór energii skutkowałby jego brakiem (lub wartością ujemną)